torchgeo/datasets/cbf.py

```diff
@@ -108,9 +113,9 @@ def _check_integrity(self) -> bool:
         Returns:
             True if dataset files are found and/or MD5s match, else False
         """
-        assert isinstance(self.paths, str)
+        assert check_instance_type(self.paths)
```

Comment:
Seems like this dataset doesn't actually support iterables. If we change the type hints to just `Path` does mypy complain?

Reply:
@adamjstewart 

Code:
``assert isinstance(self.paths, Path)``.

Mypy error:
```
Parameterized generics cannot be used with class or instance checks  [misc]
Argument 2 to "isinstance" has incompatible type "<typing special form>"; expected "_ClassInfo"  [arg-type]
```